canonical-web-and-design/maas-ui
# fix(proxy): fix XHR HTTP requests from the dev proxy

## Done

- Fix XHR requests when using the dev proxy and a MAAS with TLS enabled.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- [Enable TLS](https://github.com/canonical-web-and-design/app-tribe/issues/847) on your MAAS.
- Run `yarn start`.
- Check that you can log in and out.
- Go to Settings -> commissioning scripts and check that you can upload a script (example scripts here: https://maas.io/docs/commissioning-script-reference).
- Run `yarn ui` and repeat the tests for log in/out and uploading a script.
- Turn off TLS on your MAAS.
- With both `yarn start` and `yarn ui` test that you can log in/out and upload a script.

## Fixes

Fixes: canonical-web-and-design/app-tribe#886.
Fixes: #3184.
